tsl/src/chunk.c

```diff
@@ -259,3 +269,57 @@ chunk_invoke_drop_chunks(Oid relid, Datum older_than, Datum older_than_type)
 
 	return num_results;
 }
+
+static bool
+chunk_is_distributed(const Chunk *chunk)
+{
+	return chunk->relkind == RELKIND_FOREIGN_TABLE;
+}
+
+Datum
+chunk_create_replica_table(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid;
+	const char *data_node_name;
+	const Chunk *chunk;
+	const Hypertable *ht;
+	const ForeignServer *server;
+	Cache *hcache = ts_hypertable_cache_pin();
+
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	GETARG_NOTNULL_OID(chunk_relid, 0, "chunk");
+	GETARG_NOTNULL_NULLABLE(data_node_name, 1, "data node name", CSTRING);
+
+	chunk = ts_chunk_get_by_relid(chunk_relid, false);
+	if (chunk == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("relation \"%s\" is not a chunk", get_rel_name(chunk_relid))));
```

Comment:
get_rel_name returns NULL if it's not a relid. So the above ereport will crash.

Reply:
@nikkhils It does not crash. It returns NULL. It is common way in our code now.

Reply:
errmsg("relation NULL is not a chunk") <- this will crash

Reply:
It does not crash. It returns an error with not very nice message. It is good to fix on a general level, but I find it helpful to get name of the relation if the OID is valid.

Reply:
@k-rus  I don't know why you don't want to fix this..

```sql
single=# select oid from pg_database where datname = 'single';
  oid  
-------
 16384
(1 row)
single=# SELECT _timescaledb_internal.create_chunk_replica_table(16384, 'data_node_1');
SSL SYSCALL error: EOF detected
The connection to the server was lost. Attempting reset: Failed.
```

here's the backtrace:

```
(lldb) bt
* thread #1, stop reason = signal SIGSTOP
  * frame #0: 0x00007fff6b9f333a libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff6baafe60 libsystem_pthread.dylib`pthread_kill + 430
    frame #2: 0x00007fff6b97a808 libsystem_c.dylib`abort + 120
    frame #3: 0x0000000100ae29bd postgres`ExceptionalCondition(conditionName="strvalue != NULL", errorType="FailedAssertion", fileName="snprintf.c", lineNumber=442) at assert.c:67:2
    frame #4: 0x0000000100b6a950 postgres`dopr(target=0x00007ffeef88b338, format="\" is not a chunk", args=0x00007ffeef88b5d0) at snprintf.c:442:4
    frame #5: 0x0000000100b6a715 postgres`pg_vsnprintf(str="relation \"\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7
```

Reply:
Thank you for demonstrating the crash. I didn't get it when I tested myself.
I will look into a general fix.

Reply:
Please add a test case for this in your patch. Thanks

Reply:
Just for the record, when I do above steps I get:
```
ERROR:  relation "(null)" is not a chunk
```

Reply:
Fixed and tested